ui/app/components/mfa-login-enforcement-form.js

```diff
@@ -0,0 +1,152 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { methods } from 'vault/helpers/mountable-auth-methods';
+import { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency';
+
+/**
+ * @module MfaLoginEnforcementForm
+ * MfaLoginEnforcementForm components are used to create and edit login enforcements
+ *
+ * @example
+ * ```js
+ * <MfaLoginEnforcementForm @model={{this.model}} @mfaMethod={{this.mfaMethod}} />
+ * ```
+ * @callback onSave
+ * @callback onClose
+ * @param {Object} model - login enforcement model
+ * @param {Object} [mfaMethod] - provide when creating a login enforcement for a selected method -- otherwise search selector is displayed
+ * @param {boolean} [hasActions] - whether the save and cancel actions will be displayed and handled internally or not
+ * @param {onSave} [onSave] - triggered on save success
+ * @param {onClose} [onClose] - triggered on cancel
+ */
+
+export default class MfaLoginEnforcementForm extends Component {
+  @service store;
+  @service flashMessages;
+
+  targetTypes = [
+    { label: 'Authentication mount', type: 'accessor', key: 'auth_method_accessors' },
+    { label: 'Authentication method', type: 'method', key: 'auth_method_types' },
+    { label: 'Group', type: 'identity/group', key: 'identity_groups' },
+    { label: 'Entity', type: 'identity/entity', key: 'identity_entities' },
+  ];
+  authMethods = methods();
+  searchSelectOptions = null;
+
+  @tracked name;
+  @tracked mfaMethods = [];
+  @tracked targets = [];
+  @tracked selectedTargetType = 'accessor';
+  @tracked selectedTargetValue = null;
+  @tracked searchSelect = {
+    options: [],
+    selected: [],
+  };
+
+  constructor() {
+    super(...arguments);
+    // aggregate different target array properties on model into flat list
+    this.flattenTargets();
+    // eagerly fetch identity groups and entities for use as search select options
+    this.resetTargetState();
+  }
+
+  async flattenTargets() {
+    for (let target of this.targetTypes) {
+      const targetArray = await this.args.model[target.key];
+      this.targets.addObjects(targetArray);
+    }
+  }
+  async resetTargetState() {
+    this.selectedTargetValue = null;
+    this.searchSelect.selected = [];
+    const options = this.searchSelectOptions || {};
+    if (!this.searchSelectOptions) {
+      try {
+        const types = ['identity/group', 'identity/entity'];
+        for (const type of types) {
+          options[type] = (await this.store.query(type, {})).toArray();
+        }
+        this.searchSelectOptions = options;
+      } catch (error) {
+        // JLR TODO - swallowing error for now
```

Comment:
I was going to defer but now I think it would be better to deal with it now. What are your thoughts on this? I wonder if it would be best to hide identity groups and entities from the targets selection if the request(s) fail for any reason since there will be nothing to select.

Reply:
@hashishaw I reworked the try catch a bit so that it will catch the errors individually and then set the variable to an empty array. I think that will be good for now to ensure nothing breaks and then I can look into a more detailed solution once I can test the full workflow more thoroughly.